### PR TITLE
Shared services setup

### LINKED ISSUE #967: [BUG] Unexpected Nexus configuration step when running `make all`
The documentation implies that at the end of the `make all` run the user will presented with the output variables and they should proceed in running `make letsencrypt`. However, after the output variables "Configuring Nexus" step is initiated.

Related bug: #968

### PR DESCRIPTION
**Pull request by deniscep**
# Shared services setup

## What is being addressed

TRE Admin should have a jumpbox available in AzureTRE deployment in order to configure shared services. This PR adds required functionalities and documentation.

## How is this addressed

- Adds a jumpbox VM to Shared Subnet and stores keys in KV
- Adds documentation on configuring shared services through jumpbox
- Modifies firewall for shared subnet to allow access to git installation, github, powershell repos, etc.
- Modifies application gateway so that Nexus and Gitea are not exposed through AGW
- Modifies Gitea config so that the root URL is a local URl
- Modifies Nexus config script so that it takes parameters instead of env variables
- Modifies Gitea repo config script so that it takes parameters instead of env variables
- Modifies InnerEye service version to match the version required by InnerEye workspace (0.1.3)

Closes #967 #975 #968 #976

### CODE REVIEW

File: scripts/gitea_migrate_repo.sh [hunk 1]
```diff
@@ -1,39 +1,62 @@
...
-tokenSecretName="gitea-"$TRE_ID"-admin-token"
-pwdSecretName="gitea-"$TRE_ID"-admin-password"
-
-gitea_url="https://$TRE_ID.$LOCATION.cloudapp.azure.com/gitea"
+keyVaultName="kv-"$tre_id
+tokenSecretName="gitea-"$tre_id"-admin-token"
+pwdSecretName="gitea-"$tre_id"-admin-password"
 
-az login --service-principal --username $ARM_CLIENT_ID --password $ARM_CLIENT_SECRET --tenant $TENANT_ID > /dev/null
+gitea_url="https://gitea-$tre_id.azurewebsites.net"
```
Comment (tompaana) [inline scripts/gitea_migrate_repo.sh:48]: Nit: Unify variable naming to be consistent: either with snake case or camel case, but not both!

File: templates/core/terraform/admin-jumpbox/vm.tf [hunk 2]
```diff
@@ -0,0 +1,79 @@
+data "azurerm_subscription" "current" {}
+
+resource "azurerm_network_interface" "jumpbox-nic" {
```
Comment (tompaana) [inline templates/core/terraform/admin-jumpbox/vm.tf:3]: `jumpbox_nic`, not `jumpbox-nic`

File: templates/core/terraform/admin-jumpbox/vm.tf [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+  os_profile_windows_config {
+  }
+
+  tags = {
+    environment = "staging"
+  }
+}
+
+resource "azurerm_key_vault_secret" "jumpbox-credentials" {
```
Comment (tompaana) [inline templates/core/terraform/admin-jumpbox/vm.tf:75]: `jumpbox_credentials`

File: templates/core/terraform/firewall/firewall.tf [hunk 4]
```diff
@@ -86,6 +86,46 @@ resource "azurerm_monitor_diagnostic_setting" "firewall" {
   }
 }
 
+resource "azurerm_firewall_application_rule_collection" "shared_subnet" {
+  name                = "arc-shared_subnet"
+  azure_firewall_name = azurerm_firewall.fw.name
+  resource_group_name = azurerm_firewall.fw.resource_group_name
+  priority            = 100
+  action              = "Allow"
+
```
Comment (tompaana) [inline templates/core/terraform/firewall/firewall.tf:95]:
```suggestion
```

File: templates/core/terraform/firewall/firewall.tf [hunk 5]
```diff
@@ -86,6 +86,46 @@ resource "azurerm_monitor_diagnostic_setting" "firewall" {
...
+
+  rule {
+    name = "admin-resources"
+    protocol {
```
Comment (tompaana) [inline templates/core/terraform/firewall/firewall.tf:99]:
```suggestion

    protocol {
```

File: templates/core/terraform/firewall/firewall.tf [hunk 6]
```diff
@@ -86,6 +86,46 @@ resource "azurerm_monitor_diagnostic_setting" "firewall" {
...
+      port = "443"
+      type = "Https"
+    }
+    protocol {
```
Comment (tompaana) [inline templates/core/terraform/firewall/firewall.tf:103]:
```suggestion

    protocol {
```

File: templates/core/terraform/firewall/firewall.tf [hunk 7]
```diff
@@ -86,6 +86,46 @@ resource "azurerm_monitor_diagnostic_setting" "firewall" {
...
+      "*githubusercontent.com",
+      "*core.windows.net",
+      "aka.ms",
+      "management.azure.com",
+      "graph.microsoft.com",
+      "login.microsoftonline.com",
+      "aadcdn.msftauth.net",
+      "graph.windows.net"
+    ]
+    source_addresses = data.azurerm_subnet.shared.address_prefixes
```
Comment (tompaana) [inline templates/core/terraform/firewall/firewall.tf:123]:
```suggestion

    source_addresses = data.azurerm_subnet.shared.address_prefixes
```

File: templates/core/terraform/firewall/firewall.tf [hunk 8]
```diff
@@ -86,6 +86,46 @@ resource "azurerm_monitor_diagnostic_setting" "firewall" {
...
+  }
+
```
Comment (tompaana) [inline templates/core/terraform/firewall/firewall.tf:125]:
```suggestion
```

File: templates/core/terraform/firewall/firewall.tf [hunk 9]
```diff
@@ -86,6 +86,46 @@ resource "azurerm_monitor_diagnostic_setting" "firewall" {
...
+}
+
+
```
Comment (tompaana) [inline templates/core/terraform/firewall/firewall.tf:128]:
```suggestion
```

File: templates/core/terraform/firewall/firewall.tf [hunk 10]
```diff
@@ -121,6 +161,11 @@ resource "azurerm_firewall_application_rule_collection" "resource_processor_subn
     source_addresses = data.azurerm_subnet.resource_processor.address_prefixes
   }
 
+  depends_on = [
+    azurerm_firewall_application_rule_collection.shared_subnet
+  ]
+
+
```
Comment (tompaana) [inline templates/core/terraform/firewall/firewall.tf:168]:
```suggestion
```

File: .github/workflows/deploy_tre.yml [hunk 11]
```diff
@@ -5,7 +5,7 @@ on:
     # 1am each night https://crontab.guru/#0_1_*_*_*
     - cron: "0 1 * * *"
   push:
-    branches: [develop, main]
+    branches: [develop, main, deniscep/innereye-test]
```
Comment (tompaana) [inline .github/workflows/deploy_tre.yml:8]: Did you forget to remove this?

File: templates/core/terraform/admin-jumpbox/vm.tf (hunk 2 above)
Comment (deniscep) [reply templates/core/terraform/admin-jumpbox/vm.tf:3]: fixed

File: templates/core/terraform/admin-jumpbox/vm.tf (hunk 3 above)
Comment (deniscep) [reply templates/core/terraform/admin-jumpbox/vm.tf:75]: fixed

File: scripts/gitea_migrate_repo.sh (hunk 1 above)
Comment (deniscep) [reply scripts/gitea_migrate_repo.sh:48]: fixed

File: .github/workflows/deploy_tre.yml (hunk 11 above)
Comment (deniscep) [reply .github/workflows/deploy_tre.yml:8]: will remove before merge;

File: templates/shared_services/gitea/terraform/gitea-webapp.tf [hunk 12]
```diff
@@ -34,7 +34,7 @@ resource "azurerm_app_service" "gitea" {
     GITEA_PASSWD   = "@Microsoft.KeyVault(SecretUri=${azurerm_key_vault_secret.gitea_password.id})"
     GITEA_EMAIL    = "giteaadmin@tre.com"
```
Comment (msebragge) [inline templates/shared_services/gitea/terraform/gitea-webapp.tf:35]: I noticed earlier that username gitea_admin did not work (giteaadmin did work). Can we align the username and email?

Comment (deniscep) [reply templates/shared_services/gitea/terraform/gitea-webapp.tf:35]: fixed

**Review by tompaana — APPROVED** [review]

**Pull request merged**